x-pack/packages/kbn-elastic-assistant/impl/assistant/api/index.test.tsx

```diff
@@ -160,7 +176,11 @@ describe('API tests', () => {
 
       const result = await fetchConnectorExecuteAction(fetchConnectorArgs);
 
-      expect(result).toEqual({ response: API_ERROR, isStream: false, isError: true });
+      expect(result).toEqual({
+        response: `${API_ERROR}\n\nCould not get reader from response`,
+        isStream: false,
+        isError: true,
+      });
     });
 
     it('returns API_ERROR + error message on non streaming responses', async () => {
```

Comment:
Seeing a bug when using bedrock with bad credentials and streaming is disabled. The error returned immediately disappears and isn't persisted:



https://github.com/elastic/kibana/assets/2946766/f1c826be-6dc4-4504-a8b0-5f5d70ff7e7d

Reply:
known bug to do with persistence, @YulNaumenko is addressing separately

Reply:
Sounds good -- resolving this one then 👍

Reply:
Yep, we are not persisting error responses. Will fix it after FF.